### PR TITLE
 Added lowercase split value and passing tests#42

### LINKED ISSUE #42: Use lower case names for split values
**Is your feature request related to a problem? Please describe.**
Change split values from all caps to lower case. 
This makes file/directory naming more consistent with the split.

**Describe the solution you'd like**
TRAIN -> train
VALIDATION -> validation
TEST -> test

**Describe alternatives you've considered**
1. No change
- There's a bit of skew when it comes to mapping split value with file/dir name.

### PR DESCRIPTION
**Pull request by Samad999777**
This pull request takes care of the lower case support issue and enables the users to follow the normal convention as mentioned in #42

This is mainly a cosmetic change and the dataset labels such as 'TRAIN' ,'TEST', 'VALIDATE' are not compatible as lower cases versions. It does not affect the broader scope of the project. however it may require documentation changes.

Fixes  #42 

Cosmetic change. as explained in summary above.

Please delete options that are not relevant.

- [ Y ] New feature (non-breaking change which adds functionality)
- [ Y ] This change requires a documentation update

## Checklist

- [Y] My code adheres to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html)
- [Y] I ran `make test` and all tests pass
- [Y] I ran the tool and verified the change works
- [Y] I have adequately commented my code, particularly in hard-to-understand areas
- [Y] My changes generate no new warnings
- [Y] I have corrected any misspellings in my code
- [Y] (For feature) I made sure that I'm merging to the `dev` branch

### CODE REVIEW

File: tfrecorder/types.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ class ImageUri(SupportedType):
 class SplitKey(SupportedType):
   """Supports split key columns."""
   feature_spec = tf.io.FixedLenFeature([], tf.string)
-  allowed_values = ['TRAIN', 'VALIDATION', 'TEST', 'DISCARD']
+  allowed_values = ['train', 'validation', 'test', 'discard']
```
Comment (cfezequiel) [inline tfrecorder/types.py:49]: You should also update any code that uses `DISCARD` and replace it with `discard`.

File: tfrecorder/utils_test.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def test_valid_records(self, mock_fn):
 
     mock_fn.return_value = {self.split: self.dataset}
     num_records = len(self.data['image'])
-
+    #omit dir='/tmp' if on windows
```
Comment (cfezequiel) [inline tfrecorder/utils_test.py:78]: You can remove `dir='/tmp'`. It should work on Linux as well.

**Review by cfezequiel — CHANGES_REQUESTED** [review]
Thanks for making a new PR with the changes. I added some comments.
Please make sure to pass all the tests, including the ones for Jupyter notebooks.

File: tfrecorder/utils_test.py (hunk 2 above)
Comment (Samad999777) [reply tfrecorder/utils_test.py:78]: @cfezequiel Thanks for the comments,
I'll work on these changes and update the code, my apologies repeatedly miaaing out on things, hopefully after this first round of changes. I will understand the package better and things will become much smoother from here and on.

File: tfrecorder/types.py (hunk 1 above)
Comment (Samad999777) [reply tfrecorder/types.py:49]: @cdibona I edited the files too, but one running some tests. I am getting this error.
`    raise NotImplementedError(_DECODE_DEPRECATION_MESSAGE)
NotImplementedError: TFXIO should be used to decode CSV.`
Can you help me out? I couldn't find help on it online.

**Comment by Samad999777** [comment]
@cfezequiel kindly review the new commits

File: samples/Basic-TFRecorder-Usage.ipynb [hunk 3]
```diff
@@ -26,17 +36,17 @@
...
   {
    "cell_type": "code",
-   "execution_count": 2,
+   "execution_count": 3,
    "metadata": {},
    "outputs": [],
    "source": [
     "input_file = pathlib.Path(os.getcwd())/'../tfrecorder/test_data/data.csv'\n",
-    "output_dir = './out'  # YOUR LOCAL OUTPUT PATH HERE"
+    "output_dir = 'D:\\Open Source\\Google\\out'  # YOUR LOCAL OUTPUT PATH HERE"
```
Comment (cfezequiel) [inline samples/Basic-TFRecorder-Usage.ipynb:44]: Please change the `output_dir` value to something simpler, like `output`. The current path will not expand correctly in a Unix-based system.

Also at the end of the notebook, consider adding a cell to delete the generated output directory.

File: samples/Basic-TFRecorder-Usage.ipynb [hunk 4]
```diff
@@ -139,75 +149,20 @@
...
-      "text/plain": [
-       "{'job_id': 'DirectRunner',\n",
-       " 'metrics': {'rows': 6, 'good_images': None, 'bad_images': 6},\n",
-       " 'tfrecord_dir': './out/tfrecorder-20201028-160301-to-tfr'}"
-      ]
-     },
-     "execution_count": 4,
-     "metadata": {},
-     "output_type": "execute_result"
+     "ename": "FileNotFoundError",
```
Comment (cfezequiel) [inline samples/Basic-TFRecorder-Usage.ipynb:152]: I'd suggest clearing the outputs of the notebooks before you check them in to Git.

**Review by cfezequiel — COMMENTED** [review]
Thanks for updating the notebooks. I added a few comments.

File: samples/Basic-TFRecorder-Usage.ipynb (hunk 4 above)
Comment (Samad999777) [reply samples/Basic-TFRecorder-Usage.ipynb]: All done and have pushed the changes

File: samples/Basic-TFRecorder-Usage.ipynb (hunk 3 above)
Comment (Samad999777) [reply samples/Basic-TFRecorder-Usage.ipynb]: done

**Comment by Samad999777** [comment]
@cfezequiel Kindly review the changes

**Comment by Samad999777** [comment]
@cfezequiel Did you review the changes ?

File: tfrecorder/test_data/data.csv [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
-TRAIN,tfrecorder/test_data/images/TRAIN/cat/cat-640x853-1.jpg,cat
-TRAIN,tfrecorder/test_data/images/TRAIN/goat/goat-640x640-1.jpg,goat
-VALIDATION,tfrecorder/test_data/images/VALIDATION/cat/cat-800x600-2.jpg,cat
-VALIDATION,tfrecorder/test_data/images/VALIDATION/goat/goat-320x320-2.jpg,goat
+test,tfrecorder/test_data/images/TEST/cat/cat-800x600-3.jpg,cat
+test,tfrecorder/test_data/images/TEST/goat/goat-640x427-3.jpg,goat
+train,tfrecorder/test_data/images/TRAIN/cat/cat-640x853-1.jpg,cat
+train,tfrecorder/test_data/images/TRAIN/goat/goat-640x640-1.jpg,goat
+validation,tfrecorder/test_data/images/VALIDATION/cat/cat-800x600-2.jpg,cat
+validation,tfrecorder/test_data/images/VALIDATION/goat/goat-320x320-2.jpg,goat
```
Comment (cfezequiel) [inline tfrecorder/test_data/data.csv]:
```suggestion
test,tfrecorder/test_data/images/test/cat/cat-800x600-3.jpg,cat
test,tfrecorder/test_data/images/test/goat/goat-640x427-3.jpg,goat
train,tfrecorder/test_data/images/train/cat/cat-640x853-1.jpg,cat
train,tfrecorder/test_data/images/train/goat/goat-640x640-1.jpg,goat
validation,tfrecorder/test_data/images/validation/cat/cat-800x600-2.jpg,cat
validation,tfrecorder/test_data/images/validation/goat/goat-320x320-2.jpg,goat
```

**Review by cfezequiel — CHANGES_REQUESTED** [review]
@Samad999777 sorry for the long delay in response. Just got back to reviewing this PR and also testing your changes. I added one suggestion. Once you resolve it I think this should be good to check in.

**Comment by Samad999777** [comment]
@cfezequiel alright,
I will make the changes and hopefully we can publish these last changes as well.

**Comment by Samad999777** [comment]
@cfezequiel can you kindly review again and approve ?

**Comment by cfezequiel** [comment]
@Samad999777 looks like there is a conflict with one of the files. Could you please check?

**Comment by Samad999777** [comment]
@cfezequiel sure, i'll review

**Comment by Samad999777** [comment]
@cfezequiel I've resolved the conflict. Can you review ?

**Comment by cfezequiel** [comment]
@Samad999777 got it. There seems to be a cla issue but it looks like something I need to resolve on my end.

**Comment by Samad999777** [comment]
@cfezequiel yes, i think you need to re-submit your CLA.

**Comment by Samad999777** [comment]
@cfezequiel can you kindly take care of the CLA so these changes can be pushed?

**Comment by cfezequiel** [comment]
@Samad999777 I may need to update the commit in your PR and push the change to your branch to address the CLA issue (basically remove the unregistered email address there). Are you ok with that?

**Comment by Samad999777** [comment]
@cfezequiel As long as i get to be the contributer of this pull request in the project. I am very much okay with it.

**Comment by cfezequiel** [comment]
@Samad999777 are you able to revert this commit: "Merge branch 'dev' into main" and do a rebase against the `dev` branch instead? That way, your commits come out on top in the commit history.

**Comment by Samad999777** [comment]
@cfezequiel I have reverted the commit: "Merge branch 'dev' into main". However when i attempted to rebase it says that it is already upto date [see image below]. Am i doing this right ?
image

**Comment by cfezequiel** [comment]
@Samad999777 Thanks for the change. Have you tried rebasing the main branch in your fork with this repo's dev branch? PRs should be made off of the dev branch unless it's a hotfix.

**Pull request closed without merging**